apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/SignedUrlUtilsTest.cs

```diff
@@ -32,108 +30,148 @@ namespace Google.Storage.V1.IntegrationTests
+                    Assert.Null(obj);
+
+                    // Restore the object. 
+                    await _fixture.Client.UploadObjectAsync(bucket, name, "", new MemoryStream(_fixture.SmallContent));
+                },
+                afterDelay: async () =>
+                {
+                    // Verify that the URL no longer works.
+                    var response = await _fixture.HttpClient.DeleteAsync(url);
+                    Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+                    var obj = await _fixture.Client.ListObjectsAsync(bucket, name).FirstOrDefault(o => o.Name == name);
+                    Assert.NotNull(obj);
+
+                    // Cleanup
+                    await _fixture.Client.DeleteObjectAsync(bucket, name);
+                });
         }
 
         [Fact]
         public async void GetTest()
         {
-            var url = SignedUrlUtils.Create(
-                _fixture.ReadBucket,
-                _fixture.SmallObject,
-                _credentialsFilePath,
-                _duration);
-
-            // Verify that the URL works initially.
-            var response = await _httpClient.GetAsync(url);
-            var result = await response.Content.ReadAsByteArrayAsync();
-            Assert.Equal(_fixture.SmallContent, result);
+            await _fixture.FinishDelayTest(GetTestName(nameof(GetTest)));
+        }
 
-            // Wait until the URL expires.
-            await Task.Delay(_duration);
+        private void GetTest_InitDelayTest()
+        {
+            string url = null;
 
-            // Verify that the URL no longer works.
-            response = await _httpClient.GetAsync(url);
-            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            _fixture.RegisterDelayTest(GetTestName(nameof(GetTest)), _duration,
+                beforeDelay: async duration =>
+                {
+                    url = SignedUrlUtils.Create(
+                        _fixture.ReadBucket,
+                        _fixture.SmallObject,
+                        _fixture.CredentialsFilePath,
+                        duration);
+
+                    // Verify that the URL works initially.
+                    var response = await _fixture.HttpClient.GetAsync(url);
+                    var result = await response.Content.ReadAsByteArrayAsync();
+                    Assert.Equal(_fixture.SmallContent, result);
+                },
+                afterDelay: async () =>
+                {
+                    // Verify that the URL no longer works.
+                    var response = await _fixture.HttpClient.GetAsync(url);
+                    Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+                });
         }
 
         [Fact]
         public async void GetBucketTest()
         {
-            var bucket = _fixture.ReadBucket;
+            await _fixture.FinishDelayTest(GetTestName(nameof(GetBucketTest)));
```

Comment:
Done (CallerMemberName and Task). Good idea about using attributes, but I think you're right that this can be done later. I'll see if I can find some time later today to try this out and see if things can be cleaned up here.